### PR TITLE
Implementation of File System Repository

### PR DESCRIPTION
**Pull request by tomstich**

### CODE REVIEW

**Comment by Partyschaum** [comment]
I fixed the broken test. Please merge this PR only **after** you finished all items from the issue.

**Comment by Partyschaum** [comment]
See my commit comments. Recursion FTW! If you are interested and find the time after working on the issue have a look the the implementation of `deleteRecursive($file)` and learn. 😄

File: tests/ReportFileRepositoryTest.php [hunk 1]
```diff
@@ -138,6 +136,21 @@ public function itShouldFindByStatus()
...
+    /**
+     * @test
+     */
+    public function itShouldFindById()
+    {
+        $repository = new ReportFileRepository(self::REPORTS_ROOT_PATH);
+        $content = 'some content';
+
+        $report1 = $repository->create(uniqid(), $content);
+        $report2 = $repository->create(uniqid(), $content);
```
Comment (Partyschaum) [inline tests/ReportFileRepositoryTest.php]: What happens with `$report2`?

Comment (SharktheFire) [inline tests/ReportFileRepositoryTest.php]: Forget to delete the line of **$report2**. This should be inside of the third issue.

Comment (Partyschaum) [inline tests/ReportFileRepositoryTest.php]: You could also leave it and have a second assertion. 😸

**Pull request merged**